### ISSUE TITLE
PixelWatchFace is no longer on the Play Store

### ISSUE THREAD

**Issue by yashdavisgupta** [CorvetteCole/PixelWatchFace#68]
PixelWatchFace is no longer on the Play Store

**Comment by CorvetteCole** [comment]
working on it, google removed my application for not adequately explaining why it needed location permission (I previously explained it used location to get the weather).

**Comment by yashdavisgupta** [comment]
Cool, thanks for the update.

**Comment by KayriLynn** [comment]
I just switched phones and looked for your app on the Play Store and couldn't find it with my new phone, old phone, watch or computer. Is this an ongoing issue? I'm sad, because your watch face is my favorite!

**Comment by ELY3M** [comment]
why not put it on fdroid.org?

**Comment by CorvetteCole** [comment]
it isn't on f-droid because unfortunately I rely on some Google wearable APIs to actually transmit data between the watch and phone. I am going to fix the play store listing, and later I may complete the rewrite of the phone settings.

I have been busy working at an aerospace company, so I haven't had much time for this project but I will try to actually get to it rather than pushing it off

**Comment by michaelteck** [comment]
Thank you, @CorvetteCole , this is my favorite watch face and, since upgrading my phone, I am anxiously waiting to load the face back to my watch.

**Comment by CorvetteCole** [comment]
I'm gonna give you all a timeline instead of leaving it hanging. I'll have this back on the Play Store by July 4th! Will try to get back in to this project after it is on the Play Store. I did have some cool ideas I wanted to finish before I got burned out

**Comment by michaelteck** [comment]
> I'm gonna give you all a timeline instead of leaving it hanging. I'll have this back on the Play Store by July 4th! Will try to get back in to this project after it is on the Play Store. I did have some cool ideas I wanted to finish before I got burned out

I'll go out on a limb on behalf of others: We greatly appreciate it!

**Comment by yashdavisgupta** [comment]
@CorvetteCole If you create issues that detail what you want to see, I can help implement some features. Excited to see it back on the play store as well.

**Comment by CorvetteCole** [comment]
@yashdavisgupta I can publish some of my technical drawings and plans after it's back on the Play store. 

basically the phone app will be rewritten in flutter, with a new architecture where watch settings are stored on the watch as an authoritative source. This makes the companion application a client to the watch "server". 

The watch will store the available settings and setting structure and the phone app will pull those to generate settings in the app. This allows me to push new options without updating the phone app. Additionally, settings will be available on the watch face as well with this.

After this architectural rewrite I'll be able to support adding new features quickly such as complications and material you support (I want to pull colors from the phone to match)

I've also got a few big plans regarding a new weather app I've been working on which I'll announce later once this project is successfully revived

**Comment by CorvetteCole** [comment]
...currently working at a space company writing software that runs on the ISS so I'll have to do this in a sustainable fashion (to prevent burnout again)

**Comment by CorvetteCole** [comment]
and we're off... changes sent for review. The app was already 99% compliant I think this was a mixture of burnout and becoming very busy very fast. Thanks for the patience of everybody here. By the way, I will be rebranding soon from cole.dev to UserFriendly.dev since I recently managed to purchase that domain (and cole.dev hyperlinks to another guys website which really brings me down :(  ).

I will update this thread when I get news back

**Comment by CorvetteCole** [comment]
Honestly for not being touched in nearly 2 years the code held up rather well. I'm not really criticizing myself there too much. Definitely some rewrite tasks to be done a little on the watch face and the phone app needs a full rehaul. I guess I'll finally be able to use "2.0" as a release version!

**Comment by CorvetteCole** [comment]
update: rejected again, although I think this is a bug in their system since it is reviewing the wrong artifact. will work on this later today

**Comment by CorvetteCole** [comment]
update: this weekend I am submitting a new update which should hopefully be approved. I had to update a few APIs and had one potential pendingintent vulnerability. This should be back online soon!

**Comment by michaelteck** [comment]
Hey @CorvetteCole! I hope all is well with you and you are just busy with more important things. Is there anything new for us fans of your Pixel Watch Face?

**Comment by CorvetteCole** [comment]
Yeah, sorry, I'm pretty frustrated right now. Have had a few updates rejected. I've also simultaneously gotten insanely busy elsewhere. I've been working on a secret project to allow me to write watch faces in Flutter (I will release it independently). Basically the plan is to use this vastly simplified dev process to entirely rewrite both the watch face and companion app, and integrate Material You and all the watch styles brought by the actual upcoming Pixel Watch. I don't have a timeline unfortunately right now. Working part-time + university + some other things tends to be quite time consuming! 

TL;DR: still working on getting current version back to play store, just busy. working on rewrite in fancy framework that doesn't actually support what I'm doing with it (I'm building that part).

It helps that I'm not quite burned out on this anymore so I feel motivated to do these things, especially via Flutter since I love it. If the Flutter version isn't performant enough or too much of a battery hog I'll drop it so don't worry, still testing prototypes.
